src/query/storages/system/src/tables_table.rs

```diff
@@ -133,128 +133,9 @@ where TablesTable<WITH_HISTORY, WITHOUT_VIEW>: HistoryAware
-                                (ctl_name.to_string(), database_name.to_string(), name)
-                            }));
-                        }
-                        Err(err) => {
-                            warn!(
-                                "Failed to list table names in database {}.{}: {}",
-                                ctl_name, database_name, err
-                            );
-                            ctx.push_warning(format!(
-                                "Failed to list table names in database {}.{}: {}",
-                                ctl_name, database_name, err
-                            ));
-                        }
-                    }
-                }
-            }
-            let catalogs: Vec<String> = all_table_names
-                .iter()
-                .map(|(catalog_name, _, _)| catalog_name.clone())
-                .collect();
-            let databases: Vec<String> = all_table_names
-                .iter()
-                .map(|(_, database_name, _)| database_name.clone())
-                .collect();
-            let names: Vec<String> = all_table_names
-                .iter()
-                .map(|(_, _, table_name)| table_name.clone())
-                .collect();
-            let rows = names.len();
-
-            Ok(DataBlock::new_from_columns(vec![
-                StringType::from_data(catalogs),
-                StringType::from_data(databases),
-                UInt64Type::from_data(vec![0; rows]),
-                StringType::from_data(names),
-                UInt64Type::from_data(vec![0; rows]),
-                UInt64Type::from_data(vec![0; rows]),
-                StringType::from_data(vec![""; rows]),
-                StringType::from_data(vec![""; rows]),
-                StringType::from_data(vec![""; rows]),
-                StringType::from_data(vec![""; rows]),
-                StringType::from_data(vec![""; rows]),
-                TimestampType::from_data(vec![0; rows]),
-                TimestampType::from_opt_data(vec![Some(0); rows]),
-                TimestampType::from_data(vec![0; rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                UInt64Type::from_opt_data(vec![Some(0); rows]),
-                StringType::from_opt_data(vec![Some(""); rows]),
-                StringType::from_data(vec![""; rows]),
-                StringType::from_data(vec!["BASE TABLE"; rows]),
-            ]))
+        // Optimization target:  Fast path for known iceberg catalog SHOW TABLES
+        if self.is_iceberg_show_tables_query(&push_downs, &catalogs)? {
+            self.get_iceberg_table_names(ctx, catalogs).await
```

Comment:
It is not a good design to implement `iceberge`'s special logic, perhaps we should define a `trait` to handle some internal implement for different catalogs.